src/stores/AsyncStoreWithClient.ts

```diff
@@ -35,11 +35,6 @@ export abstract class AsyncStoreWithClient<T extends Object> extends AsyncStore<
         await this.onAction(payload);
 
         if (payload.action === 'MatrixActions.sync') {
-            // Filter out anything that isn't the first PREPARED sync.
-            if (!(payload.prevState === 'PREPARED' && payload.state !== 'PREPARED')) {
-                return;
-            }
-
```

Comment:
this will mean that people with large accounts will never see the app. The room list intentionally does not listen to the early sync as it results in a bunch of continuous and nonsensical updates.

Reply:
Ah - I thought this was what you said had been cargoculted. So this bit should just be there as it currently is? (Shouldn't RoomList be worrying about its own optimisations?)

Reply:
it's cargoculted in that it's copy/pasted without any additional documentation.

It's an optimization for all the stores that use this as well.

Reply:
OK - I'm generally finding this very hard to understand, so I've updated the comment with one that hopefully explains why the code is necessary - is this accurate?